Report ZuulExceptions through logger in ZuulServletFilter

ghenkes opened the pull request:
In ZuulServletFilter ZuulExceptions are just getting printed rather than getting added to the log like in ZuulServlet.

spencergibb commented on zuul-core/src/main/java/com/netflix/zuul/filters/ZuulServletFilter.java:
```diff
@@ -109,7 +113,7 @@ void init(HttpServletRequest servletRequest, HttpServletResponse servletResponse
         RequestContext.getCurrentContext().setThrowable(e);
         zuulRunner.error();
-        e.printStackTrace();
+        LOG.error(e.getMessage(), e);
```
My preference would be to remove `e.printStackTrace()` altogether and let the 'error' filters determine if they want to log the error.

NiteshKant replied:
+1

ghenkes commented:
@spencergibb @NiteshKant removed logging the exception in ZuulServletFilter. Also removed similar line in ZuulServlet. Added logging to the example ErrorResponse filter in zuul-netfilx-webapp.

NiteshKant commented:
Thanks @ghenkes !
